source/speech/speech.py

```diff
@@ -3093,3 +3094,47 @@ def clearTypedWordBuffer() -> None:
 	complete the word (such as a focus change or choosing to move the caret).
 	"""
 	_curWordChars.clear()
+
+
+def getSpeechSequenceWithLangs(speechSequence: SpeechSequence):
+	"""Get a speech sequence with the language description for each non default language of the read text.
+
+	:param speechSequence: The original speech sequence.
+	:return: A speech sequence containing descriptions for each non default language, indicating if the language is not supported by the current synthesizer.
+	"""
+	if not config.conf["speech"]["reportLanguage"]:
+		return speechSequence
+	filteredSpeechSequence = list()
+	for index, item in enumerate(speechSequence):
+		if (
+			not isinstance(item, LangChangeCommand)
+			or item.isDefault
+			or index == len(speechSequence) - 1
+			or item.lang == _speechState.lastReportedLanguage
+		):
+			filteredSpeechSequence.append(item)
+			continue
+		langDesc = languageHandler.getLanguageDescription(item.lang)
+		if langDesc is None:
+			langDesc = item.lang
+		filteredSpeechSequence.append(LangChangeCommand(None))
+		filteredSpeechSequence.append(langDesc)
+		_speechState.lastReportedLanguage = item.lang
+		if not languageIsSupported(item.lang):
+			# Translators: Reported when the language of the text been read is not supported.
+			filteredSpeechSequence.append("not supported")
+		filteredSpeechSequence.append(item)
+	return filteredSpeechSequence
+
+
+def languageIsSupported(language: str | None) -> bool:
```

Comment:
this function should probably be added to `SynthDriver`
e.g. `SynthDriver.languageIsSupported()` uses `self.availableLanguages`